You will need to generate a private key at Mashape, then you can use it here.

In [5]:
import unirest

# I keep mine in another file, which is not sent to GitHub
# Or put yours in the string 'key' below
try:
    from secrets import mashape as key
except ImportError as e:
    key = " KEY GOES HERE "

## Face recognition

Face++ has a comprehensive API, some of which is free. [Their site](http://www.faceplusplus.com/) allows you to build apps to do all kinds of clever things.

Let's try a simple detection on this face...

<img src="http://csegrecorder.com/assets/images/interviews/2013-10-Evan_Bianco.jpg" width="400px" />

In [6]:
url = "https://faceplusplus-faceplusplus.p.mashape.com/detection/detect"
target = "http://csegrecorder.com/assets/images/interviews/2013-10-Evan_Bianco.jpg"
params = {'url': target,
          'attribute': 'age,glass,gender,pose,race,smiling',  # optional
         }
headers={
    "X-Mashape-Key": key,
    "Accept": "application/json"
  }

# attribute
# Pass optional parameter 'attribute' to ask for age, gender, race, pose,
# glass, smiling. 'Glass' is whether the subject is wearing glasses.
# Can also pass 'all' for all, or omit for age, gender, race, and smiling.

response = unirest.post(url, headers=headers, params=params)
response.body

{u'face': [{u'attribute': {u'age': {u'range': 5, u'value': 29},
    u'gender': {u'confidence': 99.9983, u'value': u'Male'},
    u'glass': {u'confidence': 99.9624, u'value': u'None'},
    u'pose': {u'pitch_angle': {u'value': 2e-05},
     u'roll_angle': {u'value': -0.049369},
     u'yaw_angle': {u'value': 1.647306}},
    u'race': {u'confidence': 99.9588, u'value': u'White'},
    u'smiling': {u'value': 8.15481}},
   u'face_id': u'e4595d871e098bbdadca1919b814d95b',
   u'position': {u'center': {u'x': 57.0, u'y': 55.745721},
    u'eye_left': {u'x': 48.362167, u'y': 44.554523},
    u'eye_right': {u'x': 65.575167, u'y': 44.532763},
    u'height': 51.833741,
    u'mouth_left': {u'x': 48.9755, u'y': 72.201222},
    u'mouth_right': {u'x': 65.4475, u'y': 72.355501},
    u'nose': {u'x': 58.262167, u'y': 59.061369},
    u'width': 35.333333},
   u'tag': u''}],
 u'img_height': 750,
 u'img_id': u'72195bb5b852819a786c35937d0ee85d',
 u'img_width': 1100,
 u'session_id': u'6fb3a8129f514b16bd6d18da8d8e440e'

<img src="https://media.licdn.com/mpr/mpr/shrinknp_400_400/p/4/005/074/166/32433e7.jpg" />

In [7]:
target = "https://media.licdn.com/mpr/mpr/shrinknp_400_400/p/4/005/074/166/32433e7.jpg"
params = {'url': target,
          'attribute': 'age,glass,gender,pose,race,smiling',  # optional
         }

response = unirest.post(url, headers=headers, params=params)
response.body

{u'face': [{u'attribute': {u'age': {u'range': 8, u'value': 31},
    u'gender': {u'confidence': 99.9627, u'value': u'Male'},
    u'glass': {u'confidence': 99.0951, u'value': u'Normal'},
    u'pose': {u'pitch_angle': {u'value': -0.001311},
     u'roll_angle': {u'value': -15.8251},
     u'yaw_angle': {u'value': 29.297885}},
    u'race': {u'confidence': 99.11149999999999, u'value': u'White'},
    u'smiling': {u'value': 82.434}},
   u'face_id': u'0f21a53f7268b854e3976eaf09d01510',
   u'position': {u'center': {u'x': 62.664908, u'y': 42.612137},
    u'eye_left': {u'x': 54.819789, u'y': 37.731135},
    u'eye_right': {u'x': 67.791557, u'y': 34.054354},
    u'height': 31.926121,
    u'mouth_left': {u'x': 54.391029, u'y': 52.41029},
    u'mouth_right': {u'x': 70.550132, u'y': 48.505805},
    u'nose': {u'x': 67.058575, u'y': 44.52058},
    u'width': 31.926121},
   u'tag': u''}],
 u'img_height': 379,
 u'img_id': u'0632068f21deffbad60252fecf2dcb90',
 u'img_width': 379,
 u'session_id': u'0e9b06d94462

<img src="http://i.telegraph.co.uk/multimedia/archive/02788/couple_2788635b.jpg" />

In [8]:
target = "http://i.telegraph.co.uk/multimedia/archive/02788/couple_2788635b.jpg"
params = {'url': target}
response = unirest.post(url, headers=headers, params=params)
response.body

{u'face': [{u'attribute': {u'age': {u'range': 7, u'value': 39},
    u'gender': {u'confidence': 99.932, u'value': u'Male'},
    u'race': {u'confidence': 89.443, u'value': u'Asian'},
    u'smiling': {u'value': 19.8494}},
   u'face_id': u'47972d435640f62576ea707c6e8170b6',
   u'position': {u'center': {u'x': 58.583333, u'y': 20.989305},
    u'eye_left': {u'x': 56.209833, u'y': 17.985241},
    u'eye_right': {u'x': 60.984667, u'y': 16.166925},
    u'height': 19.518717,
    u'mouth_left': {u'x': 57.135333, u'y': 27.621658},
    u'mouth_right': {u'x': 62.063833, u'y': 26.189225},
    u'nose': {u'x': 57.036333, u'y': 23.090455},
    u'width': 12.166667},
   u'tag': u''},
  {u'attribute': {u'age': {u'range': 9, u'value': 31},
    u'gender': {u'confidence': 78.3792, u'value': u'Female'},
    u'race': {u'confidence': 99.7239, u'value': u'White'},
    u'smiling': {u'value': 34.9148}},
   u'face_id': u'862bafa51318ed14b73e8558a1474e1f',
   u'position': {u'center': {u'x': 45.75, u'y': 34.358289},
   

## Daydreaming

This is completely made up.

<!--
Here's the reault I want:

"""
{u'seismic': [{u'attribute': {u'f_peak': {u'range': 7, u'value': 39},
    u'wavelet': {u'confidence': 89.932, u'value': u'ricker'},
    u'phase': {u'error': 12.23421, u'value': 1.73568},
    u'fresnel': {u'error': 400, u'value': 1750},
    u'colourbar': {u'confidence': 98.53456, u'value': u'black-red'}},
    u'facies': [{u'center': {u'x': 58.583333, u'y': 20.989305},
      u'name': 'parallel',
      u'confidence': 76.45311},
    {u'center': {u'x': 213.556323, u'y': 202.234543},
      u'name': 'mounded',
      u'confidence': 67.3253245},
    {u'center': {u'x': 458.346354, u'y': 120.345636},
      u'name': 'noisy',
      u'confidence': 54.353466}],
 u'img_height': 387,
 u'img_id': u'524305ff333db315e583a4bdb5d03d0d',
 u'img_width': 620,
 u'session_id': u'928365a231974ae1a87ab51ca2066700',
 u'url': u'https://dl.dropboxusercontent.com/u/14965965/seismic_example.png'}
 """
 -->

<img src="https://dl.dropboxusercontent.com/u/14965965/seismic_example.png" />

In [18]:
# Don't execute this cell.
url = "https://imageclass.io/classify/seismic.json"
target = "https://dl.dropboxusercontent.com/u/14965965/seismic_example.png"
params = {'url': target,
          'time': '600,1600',    # time range in ms
          'trace': '1000,1400',  # trace range
          'dx': '25',            # trace spacing in m
          'attribute': 'frequency,facies,resolution',  # optional
         }
headers={
    "API_key": key,
    "Accept": "application/json"
  }

response = unirest.post(url, headers=headers, params=params)
response.body


{u'seismic': [{u'attribute': {u'f_peak': {u'range': 7, u'value': 39},
    u'wavelet': {u'confidence': 89.932, u'value': u'ricker'},
    u'phase': {u'error': 12.23421, u'value': 1.73568},
    u'fresnel': {u'error': 400, u'value': 1750},
    u'colourbar': {u'confidence': 98.53456, u'value': u'black-red'}},
    u'facies': [{u'center': {u'x': 58.583333, u'y': 20.989305},
      u'name': 'parallel',
      u'confidence': 76.45311},
    {u'center': {u'x': 213.556323, u'y': 202.234543},
      u'name': 'mounded',
      u'confidence': 67.3253245},
    {u'center': {u'x': 458.346354, u'y': 120.345636},
      u'name': 'noisy',
      u'confidence': 54.353466}],
 u'img_height': 387,
 u'img_id': u'524305ff333db315e583a4bdb5d03d0d',
 u'img_width': 620,
 u'session_id': u'928365a231974ae1a87ab51ca2066700',
 u'url': u'https://dl.dropboxusercontent.com/u/14965965/seismic_example.png'}
 


##For realz

I found [some code](https://gist.github.com/endolith/255291) by *[endolith](https://github.com/endolith)* to do frequency estimation.

I got it working for seismic stuff in another notebook, [Frequency_from_image](./Frequency_from_image.ipynb). Then I set up a `Flask` app on [PythonAnywhere](https://www.pythonanywhere.com), which was surprisingly easy. It's serving at http://kwinkunks.pythonanywhere.com.

It's headless — it only accepts API calls. See below for an example.

Let's get the frequency content of the seismic in [this image](https://math.berkeley.edu/~sethian/2006/Applications/Seismic/time_mig_img.jpg). I have already measured the pixel positions of region of the image that I want, in the time range 0 to 3 s, and the *x* range 8 to 16 km.

<img src="https://math.berkeley.edu/~sethian/2006/Applications/Seismic/time_mig_img.jpg" WIDTH="500px"/>

In [9]:
import requests

url = "http://kwinkunks.pythonanywhere.com/freq"
target = "https://math.berkeley.edu/~sethian/2006/Applications/Seismic/time_mig_img.jpg"
params = {'url': target,
          'method': 'xing',
          'tmin': 0.0,
          'tmax': 3.0,
          'region': '199,141,1608,1087',
          'avg': 'trim'
         }
headers={
    "Accept": "application/json"
  }

response = requests.get(url, headers=headers, params=params)
response.json()

{u'parameters': {u'avg': u'trim',
  u'method': u'xing',
  u'region': u'199,141,1608,1087',
  u'url': u'https://math.berkeley.edu/~sethian/2006/Applications/Seismic/time_mig_img.jpg'},
 u'result': {u'freq': {u'avg': 20.55502798329521,
   u'n': 10,
   u'sd': 1.1240551094088058},
  u'greyscale': True,
  u'img_height': 1409,
  u'img_width': 946,
  u'time range': [0.0, 3.0]}}